controllers/authpolicy_authconfig.go

```diff
@@ -112,39 +112,39 @@ func (r *AuthPolicyReconciler) desiredAuthConfig(ctx context.Context, ap *api.Au
 	authConfig.Spec.Hosts = hosts
 
 	// named patterns
-	if namedPatterns := ap.Spec.NamedPatterns; len(namedPatterns) > 0 {
+	if namedPatterns := ap.GetNamedPatterns(); len(namedPatterns) > 0 {
```

Comment:
> I have a thought/concern, which I am not sure what the best resolution is. There is nothing stop someone from accessing `Spec.NamedPatterns` directly. This could lead to errors later. One way would be to force access through methods by making the fields in the Spec private. In this case the methods would want to be on the Spec struct and not the AP struct. Access would then look like `ap.Spec.GetNamedPatterns()`.
> 
> This would need to apply to all data fields in the Spec and not just the `NamedPatterns`. It should also apply to the spec in #456.

Hmm, I dont think this will quite work as intended. If we make fields private / not exported, I guess we need setters for all the fields also, while also having setters for the defaults fields. The will work for using the struct via code.

The other one is since the field is not exported, the json field will not be available via `kubectl get policy x -o json` commands which I think makes this not viable unless I'm missing something.  

> 
> What do you think? Personal if I have access to the fields of a struct at some point I will access them directly.

Yeah, unfortunately I think is an artefact of having multiple pathways of achieving the same result. I guess the assumption is that if people are acessing the fields directly via code, they know which pathway they are using, either `spec.X` or `spec.defaults.x` but the getters will be the best way to get the data without needing to know which pathway is used